diff --git CMakeLists.txt CMakeLists.txt
index 4cef919a8..ff1ed2f8d 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -91,6 +91,18 @@ option(QT_STATIC "Link static build of QT into the binaries" FALSE)
 
 option(OPENMW_UNITY_BUILD "Use fewer compilation units to speed up compile time" FALSE)
 
+#Overwrite git hashes
+option(OVERWRITE_HASHES "overwrite git commit hashes" OFF)
+
+IF( OVERWRITE_HASHES )
+    set(OVERWRITE_COMMITHASH 0 CACHE STRING "The Commit Hash tes3mp should show")
+    message(STATUS "Commit Hash ${OVERWRITE_COMMITHASH}")
+    set(OPENMW_VERSION_COMMITHASH ${OVERWRITE_COMMITHASH})
+    set(OVERWRITE_TAGHASH 0 CACHE STRING "The Tag Hash tes3mp should show")
+    message(STATUS "Tag Hash ${OVERWRITE_TAGHASH}")
+    set(OPENMW_VERSION_TAGHASH ${OVERWRITE_TAGHASH})
+ENDIF()
+
 # what is necessary to build documentation
 IF( BUILD_DOCS )
     # Builds the documentation.
diff --git components/CMakeLists.txt components/CMakeLists.txt
index 64318f646..4be6fe8a9 100644
--- components/CMakeLists.txt
+++ components/CMakeLists.txt
@@ -4,7 +4,7 @@ project (Components)
 set (VERSION_IN_FILE "${OpenMW_SOURCE_DIR}/files/version.in")
 set (VERSION_FILE_PATH_BASE "${OpenMW_BINARY_DIR}")
 set (VERSION_FILE_PATH_RELATIVE resources/version)
-if (GIT_CHECKOUT)
+if (GIT_CHECKOUT AND NOT OVERWRITE_HASHES)
     get_generator_is_multi_config(multi_config)
     add_custom_target (git-version
        COMMAND ${CMAKE_COMMAND}
@@ -22,9 +22,9 @@ if (GIT_CHECKOUT)
             -Dgenerator_is_multi_config_var=${multi_config}
             -P ${CMAKE_CURRENT_SOURCE_DIR}/../cmake/GitVersion.cmake
             VERBATIM)
-else (GIT_CHECKOUT)
+else (GIT_CHECKOUT AND NOT OVERWRITE_HASHES)
     configure_resource_file(${VERSION_IN_FILE} ${VERSION_FILE_PATH_BASE} ${VERSION_FILE_PATH_RELATIVE})
-endif (GIT_CHECKOUT)
+endif (GIT_CHECKOUT AND NOT OVERWRITE_HASHES)
 
 if(BUILD_OPENMW OR BUILD_OPENCS)
     if (OPENGL_ES)
